solr/core/src/java/org/apache/solr/search/SolrCoreParser.java

```diff
@@ -20,19 +20,22 @@
 
 import org.apache.lucene.analysis.Analyzer;
 import org.apache.lucene.queryparser.xml.CoreParser;
+import org.apache.lucene.queryparser.xml.ParserException;
 import org.apache.lucene.queryparser.xml.QueryBuilder;
 import org.apache.lucene.queryparser.xml.builders.SpanQueryBuilder;
+import org.apache.lucene.search.spans.SpanQuery;
 import org.apache.solr.common.SolrException;
 import org.apache.solr.common.util.NamedList;
 import org.apache.solr.core.SolrResourceLoader;
 import org.apache.solr.request.SolrQueryRequest;
 import org.apache.solr.util.plugin.NamedListInitializedPlugin;
+import org.w3c.dom.Element;
 
 /**
  * Assembles a QueryBuilder which uses Query objects from Solr's <code>search</code> module
  * in addition to Query objects supported by the Lucene <code>CoreParser</code>.
  */
-public class SolrCoreParser extends CoreParser implements NamedListInitializedPlugin {
+public class SolrCoreParser extends CoreParser implements NamedListInitializedPlugin, SpanQueryBuilder {
```

Comment:
This feels wrong.  I probably get why you are doing it (to prevent changes to core lucene), but it still feels like the wrong level for this.